### ISSUE TITLE
Query: Nav expansion should have include on outermost projection

### ISSUE THREAD

**Issue by smitpatel** [aspnet/EntityFrameworkCore#16229]
```C#
        [ConditionalFact]
        public virtual void Include_on_GroupJoin_SelectMany_DefaultIfEmpty_with_coalesce_result1()
        {
            using (var context = CreateContext())
            {
                var query = from g1 in context.Gears.Include(g => g.Weapons)
                            join g2 in context.Gears
                                on g1.LeaderNickname equals g2.Nickname into grouping
                            from g2 in grouping.DefaultIfEmpty()
                            select g2 ?? g1;

                var result = query.ToList();

                Assert.Equal("Marcus", result[0].Nickname);
                Assert.Equal(2, result[0].Weapons.Count);
                Assert.Equal("Marcus", result[1].Nickname);
                Assert.Equal("Marcus", result[2].Nickname);
                Assert.Equal("Baird", result[3].Nickname);
                Assert.Equal(0, result[3].Weapons.Count);
                Assert.Equal("Marcus", result[4].Nickname);
            }
        }
```

Generated tree:
```
IQueryable<Gear> LeftJoin(
|__ outer: IQueryable<Gear> Select(
|   |__ source: DbSet<Gear>, 
|   |__ selector: (Gear g) => Include(
|           g, 
|           ICollection<Weapon> MaterializeCollectionNavigation(
|           |__ elements: IQueryable<Weapon> Where(
|           |   |__ source: DbSet<Weapon>, 
|           |   |__ predicate: (Weapon w) =>  ?= Property(g, "FullName") == Property(w, "OwnerFullName") =? ), 
|           |__ navigation: Navigation: Gear.Weapons (<Weapons>k__BackingField, ICollection<Weapon>) Collection ToDependent Weapon Inverse: Owner 4 -1 11 -1 -1), Weapons)
|__ ), 
|__ inner: DbSet<Gear>, 
|__ outerKeySelector: (Gear g1) => g1.LeaderNickname, 
|__ innerKeySelector: (Gear g2) => g2.Nickname, 
|__ resultSelector: (Gear g1 | Gear g2) => g2 ?? g1)
```

**Comment by smitpatel** [comment]
The bug is nav expansion forgot to process LeftJoin and considered it to be terminating method.

**Comment by smitpatel** [comment]
Closed in #16900
